plugins/spu2-x/src/Windows/SndOut_XAudio2.inl

```diff
@@ -86,14 +86,18 @@ private:
 
         virtual ~BaseStreamingVoice()
         {
+            DeleteCriticalSection(&cs);
         }
 
         BaseStreamingVoice(uint numChannels)
-            : m_nBuffers(Config_XAudio2.NumBuffers)
+            : pSourceVoice(nullptr)
+            , qbuffer(nullptr)
+            , m_nBuffers(Config_XAudio2.NumBuffers)
             , m_nChannels(numChannels)
             , m_BufferSize(SndOutPacketSize * m_nChannels * PacketsPerBuffer)
             , m_BufferSizeBytes(m_BufferSize * sizeof(s16))
         {
+            InitializeCriticalSection(&cs);
```

Comment:
maybe use std::mutex or atomic_flag (wrapped into a class) instead of critical section ?